clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -334,14 +334,15 @@ public static void sleep(long ms) {
      * Instantiate the class
      */
     public static <T> T newInstance(Class<T> c) {
+        if (c == null)
+            throw new NullPointerException();
```

Comment:
This won't affect the calling code as both NPE and KafkaException extends RuntimeException. And, the message "Requested class was null" was dumb. 

I did this change, as other methods in this class followed the same pattern.